sky/provision/lambda_cloud/instance.py

```diff
@@ -248,18 +249,70 @@ def query_instances(
     return statuses
 
 
-def open_ports(
-    cluster_name_on_cloud: str,
-    ports: List[str],
-    provider_config: Optional[Dict[str, Any]] = None,
-) -> None:
-    raise NotImplementedError('open_ports is not supported for Lambda Cloud')
+def open_ports(cluster_name_on_cloud: str,
+               ports: List[str],
+               provider_config: Optional[Dict[str, Any]] = None) -> None:
+    """Open ports for a cluster.
+
+    Args:
+        cluster_name_on_cloud: The cluster name (unused as rules are global).
+        ports: The ports to open.
+        provider_config: The provider config (unused).
+    """
+    del cluster_name_on_cloud, provider_config  # Unused.
+
+    # Lambda Cloud firewall rules are global (not cluster-specific)
+    # Initialize Lambda Cloud client
+    lambda_client = lambda_utils.LambdaCloudClient()
+
+    # Get existing firewall rules
+    existing_rules = lambda_client.list_firewall_rules()
+
+    # Extract existing ports
+    existing_ports: Set[int] = set()
+    for rule in existing_rules:
+        if rule.get('protocol') == 'tcp':
+            port_range = rule.get('port_range')
+            if port_range and len(port_range) == 2:
+                # If it's a single port (same min and max)
+                if port_range[0] == port_range[1]:
+                    existing_ports.add(port_range[0])
+                else:
+                    # For port ranges, add all ports in the range
+                    existing_ports.update(
+                        range(port_range[0], port_range[1] + 1))
+
+    # Convert port strings to a set of individual ports
+    ports_to_open = resources_utils.port_ranges_to_set(ports)
+
+    # Remove ports that are already open
+    ports_to_open = ports_to_open - existing_ports
+
+    # Open new ports
+    for port in ports_to_open:
+        logger.info(f'Opening port {port}/tcp')
+        try:
+            lambda_client.create_firewall_rule(port_range=[port, port],
+                                               protocol='tcp')
+        except lambda_utils.LambdaCloudError as e:
+            logger.warning(f'Failed to open port {port}: {e}')
 
 
-def cleanup_ports(
-    cluster_name_on_cloud: str,
-    ports: List[str],
-    provider_config: Optional[Dict[str, Any]] = None,
-) -> None:
-    """See sky/provision/__init__.py"""
+def cleanup_ports(cluster_name_on_cloud: str,
+                  ports: List[str],
+                  provider_config: Optional[Dict[str, Any]] = None) -> None:
+    """Skip cleanup of firewall rules.
+
+    Lambda Cloud firewall rules are global to the account, not cluster-specific.
+    We skip cleanup because rules may be used by other clusters.
```

Comment:
Does this mean we'll accumulate rules over time? 

This may be ok for now, but we should leave a TODO here. Perhaps in the future we can store the instances using the firewall rule in the "description" field of a firewall rule

Reply:
Added a TODO. Yes, it is a bit challenging, as users may also implicitly rely on some of the firewall rules, as it is global.